pkg/generate/code/set_sdk.go

```diff
@@ -718,6 +722,139 @@ func SetSDKSetAttributes(
 	return out
 }
 
+// setSDKReadMany is a special-case handling of those APIs where there is no
+// ReadOne operation and instead the only way to grab information for a single
+// object is to call the ReadMany/List operation with one of more filtering
+// fields-- specifically identifier(s). This method populates this identifier
+// field with the identifier shared between the shape and the CR. Note, in the
+// case of multiple matching identifiers, the identifier field containing 'Id'
+// will be the only field populated.
+//
+// As an example, DescribeVpcs EC2 API call doesn't have a ReadOne operation or
+// required fields. However, the input shape VpcIds field can be populated using
+// a VpcId, a field in the VPC CR's Status. Therefore, populate VpcIds field
+// with the *single* VpcId value to ensure the returned array from the API call
+// consists only of the desired Vpc.
+func setSDKReadMany(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+	op *awssdkmodel.Operation,
+	sourceVarName string,
+	targetVarName string,
+	indentLevel int,
+) string {
+	inputShape := op.InputRef.Shape
+	if inputShape == nil {
+		return ""
+	}
+
+	out := "\n"
+	indent := strings.Repeat("\t", indentLevel)
+
+	resVarPath := ""
+	pluralize := pluralize.NewClient()
+	opConfig, override := cfg.OverrideValues(op.Name)
+	shapeIdentifiers := FindIdentifiersInShape(r, inputShape)
+	var err error
+	for memberIndex, memberName := range inputShape.MemberNames() {
+		if override {
+			value, ok := opConfig[memberName]
+			memberShapeRef, _ := inputShape.MemberRefs[memberName]
+			memberShape := memberShapeRef.Shape
+			if ok {
+				switch memberShape.Type {
+				case "boolean", "integer":
+				case "string":
+					value = "\"" + value + "\""
+				default:
+					panic("Member type not handled")
+				}
+
+				out += fmt.Sprintf("%s%s.Set%s(%s)\n", indent, targetVarName, memberName, value)
+				continue
+			}
+		}
+
+		// Field renames are handled in getSanitizedMemberPath
+		resVarPath, err = getSanitizedMemberPath(memberName, r, op, sourceVarName)
+		if err != nil {
```

Comment:
OK, so `err` will be non-`nil` if `memberName` is not found in either Spec or Status of the CR, And so you check below in lines 782-794 whether there is a field in Spec or Status that is the plural (or singular) of the `memberName` (if the `memberName` is also an identifier field).

That is all good, however on line 793, you aren't handling a non-`nil` `err` returned from looking up the sanitized member path of the "flipped" memberName. Which means we're kind of flying blind in the case of odd Shapes in APIs.

I think it would be safest to `panic` after receiving a non-`nil` `err` on line 793. What do you think?

Reply:
Good catch-- my only reservation with the `panic` is **if** the following scenario is possible:

* Shape has `Names` and `Identifiers` field
* CRD model only has `Identifier` field
* We hit `Names` Shape member in this loop, flip it, and try to set `resVarPath` since it's empty string, but then we panic and exit since `Name` isn't in CRD
* One more iteration we would've hit `Identifiers` and properly mapped the field.

Have you come across any APIs like this?

Reply:
> Good catch-- my only reservation with the `panic` is **if** the following scenario is possible:
> 
> * Shape has `Names` and `Identifiers` field
> * CRD model only has `Identifier` field
> * We hit `Names` Shape member in this loop, flip it, and try to set `resVarPath` since it's empty string, but then we panic and exit since `Name` isn't in CRD
> * One more iteration we would've hit `Identifiers` and properly mapped the field.
> 
> Have you come across any APIs like this?

No, but I am no longer surprised by any AWS API :) Just being safe here.